src/api/system.c

```diff
@@ -638,6 +638,110 @@ static int f_set_window_opacity(lua_State *L) {
 }
 
 
+
+typedef struct {
+  const char* symbol;
+  void* address;
+} lua_function_node;
+
+#define P(FUNC) { "lua_" #FUNC, (void*)(lua_##FUNC) }
+static void* api_require(const char* symbol) {
+  static lua_function_node nodes[] = {
+    P(absindex), P(arith), P(atpanic), P(callk), P(checkstack), 
+    P(close), P(compare), P(concat), P(copy), P(createtable), P(dump), 
+    P(error),  P(gc), P(getallocf), P(getctx), P(getfield), P(getglobal),
+    P(gethook), P(gethookcount), P(gethookmask), P(getinfo), P(getlocal),
+    P(getmetatable), P(getstack), P(gettable), P(gettop), P(getupvalue),
+    P(getuservalue), P(insert), P(isnumber), 
+    P(isstring), P(isuserdata), P(len), P(load), 
+    P(newstate), P(newthread), P(newuserdata), P(next), 
+    P(pcallk), P(pushboolean), P(pushcclosure), 
+    P(pushfstring), P(pushinteger), P(pushlightuserdata),
+    P(pushlstring), P(pushnil), P(pushnumber), P(pushstring),
+    P(pushthread), P(pushunsigned), P(pushvalue), P(pushvfstring), P(rawequal),
+    P(rawget), P(rawgeti), P(rawgetp), P(rawlen), P(rawset), P(rawseti),
+    P(rawsetp), P(remove), P(replace), P(resume), P(setallocf), 
+    P(setfield), P(setglobal), P(sethook), P(setlocal), P(setmetatable), 
+    P(settable), P(settop), P(setupvalue), P(setuservalue), P(status), 
+    P(tocfunction), P(tointegerx), P(tolstring), P(toboolean),
+    P(tonumberx), P(topointer), P(tothread), 
+    P(tounsignedx), P(touserdata), P(type), P(typename), P(upvalueid),
+    P(upvaluejoin), P(version), P(xmove), P(yieldk)
+  };
+  for (int i = 0; i < sizeof(nodes) / sizeof(lua_function_node); ++i) {
+    if (strcmp(nodes[i].symbol, symbol) == 0)
+      return nodes[i].address;
+  }
+  return NULL;
+}
+
+static int loader_plugin(lua_State* L) {
+  size_t sname, modlen, pathlen;
+  char buffer[512] = "lua_open_";
+  const char* modname = luaL_checklstring(L, -2, &modlen);
+  const char* path = luaL_checklstring(L, -1, &pathlen);
+  void* library = SDL_LoadObject(path);
+  if (modlen == 0 || !library)
+    return luaL_error(L, "Unable to load %s: %s", modname, SDL_GetError());
+  for (sname = modlen - 1; sname > 0 && modname[sname] != '.'; --sname);
```

Comment:
its unclear that the for loop does not have a body
perhaps add an empty body or switch to a while loop?

Reply:
So, I did think this was a normal idiom in C for counting things, but I have no issues changing the `;` to `{}`. Does anyone else have any comments on this one?

Reply:
Oh it is but it comes at the cost of readability

Reply:
To be honest, I find it just as readable; again, I think it's a common idiom used frequently in C (at least C that's I've read).